app/controllers/TestUserController.php

```diff
@@ -0,0 +1,219 @@
+		Project::where('user_id', Auth::id())->delete();
+		Client::where('user_id', Auth::id())->delete();		
+		User::where('id', Auth::id())->delete();
+
+		// Logout and redirect back to home page
+		Auth::logout();
+		return Redirect::to('/');
+	}
+
+	/**
+	 * Logout the user from the application
+	 */	
+	public function logout(){
+		Auth::logout();
+		return Redirect::to('/');
+	}
+
+	/**
+	 * Login the user and start a session
+	 */
+	public function login()
+	{				
+		$email 		=	Input::get('email');
+		$password	=	Input::get('password');
+
+
+		// lets validate the users input
+		$validator = Validator::make(
+			array(
+					'email' 	=>	$email,
+					'password' 	=> 	$password
+			),
+			array(
+					'email'		=> 	'required|email',
+					'password'	=>	'required'
+			)
+		);
+
+		if ($validator->fails()){
+		    return Redirect::back()->withErrors($validator)->withInput();
+		}else{
+			if( Auth::attempt(array('email' => $email, 'password' => $password)) ){				
+				return Redirect::to('hud');
+			}else{				
+				$validator->getMessageBag()->add('input', 'Incorrect email or password');
+				return Redirect::back()->withErrors($validator)->withInput();;
+			}			
+		}
+	}	
+
+	/**
+	 * Register the user and start a session
+	 */
+	public function register()
+	{	
+		$fullName	=	Input::get('fullName');
+		$email 		=	Input::get('email');
+		$password	=	Input::get('password');	
+
+		// lets validate the users input
+		$validator = Validator::make(
+			array(
+					'fullName' 	=> 	$fullName,
+					'email' 	=>	$email,
+					'password' 	=> 	$password
+			),
+			array(
+					'fullName' 	=> 	'required',
+					'email'		=> 	'required|email|unique:users',
+					'password'	=>	'required|min:8'
+			)
+		);
+
+        // Commented out for open beta and beyond
+		/************* Make sure the email being used has been invited to beta
+         *
+		if ( !DB::table('beta')->whereEmail($email)->whereStatus(1)->get() ) {
+			$validator->getMessageBag()->add('used', 'The email used has not been invited.');
+			return Redirect::back()->withErrors($validator)->withInput();
+		}
+         *
+         ******************/
+
+		if ($validator->fails()){
+		    return Redirect::back()->withErrors($validator)->withInput();
+		}
+
+		$user 				=	new User;
+		$user->full_name 	=	$fullName;
+		$user->email 		=	$email;
+		$user->password 	=	Hash::make($password);
+
+		$user->save();	
+
+		if ( Auth::attempt(array('email' => $email, 'password' => $password)) ) {
+			sendWelcomeMail();
+			return Redirect::to('hud');
+		}
+
+		return Redirect::back()->withErrors($validator);
+	}	
+
+	public function resetPassword($id)
+	{		
+		// ----------------------------------------
+		$user = User::find(Auth::id());
+		$created = $user->tasks_created;
+		$completed = $user->tasks_completed;
+
+		if ($created == "") {
+			$created = 0;
+		}
+
+		if ($completed == "") {
+			$completed = 0;
+		}
+		// ----------------------------------------
+
+		$current_pwd	=	Input::get('current_pwd');
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "current_pwd" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://172.27.59.109:9000/coding_rules#rule_key=php%3AS1192)